### PR TITLE
feature/services

### PR DESCRIPTION
**Pull request by kalachck**

### CODE REVIEW

**Comment by Genresal** [comment]
Remove Task.FromResult from services, it unnecessary there. You can fix it in the next PR.

**Comment by Genresal** [comment]
However, the await Task.FromResult(...) part is redundant because it wraps the synchronous _mapper.Map() result in a completed Task, which doesn't provide any additional benefit in this context. It simply adds an unnecessary overhead of creating and returning a completed Task.

**Comment by Genresal** [comment]
https://stackoverflow.com/questions/19568280/what-is-the-use-for-task-fromresulttresult-in-c-sharp

**Review by Genresal — APPROVED** [review]

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            }
+
+            return _mapper.Map<List<UserDto>>(users);
+        }
+
+        public async Task<List<UserDto>> GetFollowersAsync(int userId, int amount)
+        {
+            var followers = await _userRepository.GetFollowersAsync(userId, amount);
+
+            if (followers.Count == 0)
```
Comment (v12code) [inline StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs]: Why are you check count of followers and so on?
Just return empty list

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            }
+
+            return _mapper.Map<List<UserDto>>(followers);
+        }
+
+        public async Task<UserDto> GetAsync(int id)
+        {
+            var user = await _userRepository.GetAsync(id);
+
+            if (user == null)
```
Comment (v12code) [inline StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs:51]: Remove this check and throw exception in repo

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            }
+
+            await _userRepository.DeleteAsync(existingUser);
+
+            await _postRepository.DeleteRangeAsync(_mapper.Map<List<Post>>(existingUser.Posts));
+
+            foreach (var post in existingUser.Posts)
+            {
+                await _commentRepository.DeleteRangeAsync(
+                    _mapper.Map<List<Comment>>(post.Comments));
```
Comment (v12code) [inline StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs]: It's a bad idea to call the database in every loop

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            var existingUser = await _userRepository.GetAsync(id);
+
+            if (existingUser == null)
+            {
+                throw new NotFoundException(ExceptionMessages.USER_NOT_FOUND_MESSAGE);
+            }
+
+            await _userRepository.DeleteAsync(existingUser);
+
+            await _postRepository.DeleteRangeAsync(_mapper.Map<List<Post>>(existingUser.Posts));
```
Comment (v12code) [inline StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs]: Use cascading deletion when deleting a user

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs (hunk 2 above)
Comment (kalachck) [reply StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs:51]: I think, it's not so right to throws exceptions in data layer, is it?

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs (hunk 1 above)
Comment (kalachck) [reply StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs]: Don't understand what your mean "Just return empty list"

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs (hunk 2 above)
Comment (v12code) [reply StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs:51]: You right

File: StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs (hunk 1 above)
Comment (Genresal) [reply StoreManagementService/src/PBJ.StoreManagementService.Business/Services/UserService.cs]: Just return followers without checking, even if the collection is empty. It seems like it is a possible case that a user not exists followers.

**Review by v12code — APPROVED** [review]

**Pull request merged**